### PR TITLE
Test schema org

### PR DESCRIPTION
**Pull request by jyucsiro**
First cut.

See #80

### CODE REVIEW

**Comment by jyucsiro** [comment]
Allowing for netCDF metadata to be tested, e.g. URLs passed in like so:
```
$ python nc2rdf.py -o json-ld --schema-org https://www.ngdc.noaa.gov/thredds/dodsC/arctic/Polar-APP-X_v01r01_Nhem_1400_d20160801_c20160803.nc
```

result:
```
{
    "@context": "http://schema.org/",
    "description": "The Extended AVHRR Polar Pathfinder (APP-x) version-2 Thematic Climate Data Record (CDR) includes surface temperature, surface albedo, surface and the Top Of the Atmosphere (TOA) shortwave and longwave radiative fluxes, cloud properties (amount, phase, particle size, optical depth,top pressure and temperature, surface and TOA radiative effect), and ice thickness and age. The APP-x CDR has twice daily data at local solar times of 14 and 04(02) for the Arctic(Antarctic) at a spatial resolution of 25 km for both poles.",
    "http://schema.org/identifier": "Polar-APP-X_v01r01_Nhem_1400_d20160801_c20160803.nc",
    "http://schema.org/license": "No restrictions on access or use",
    "id": "_:N1ffe69b6dd6c4ceb895f7c79d3c17b2d",
    "keywords": "EARTH SCIENCE > ATMOSPHERE > ATMOSPHERIC RADIATION > SOLAR RADIATION, EARTH SCIENCE > ATMOSPHERE > ATMOSPHERIC RADIATION > LONGWAVE RADIATION, EARTH SCIENCE > ATMOSPHERE > ATMOSPHERIC RADIATION > RADIATIVE FLUX, EARTH SCIENCE > ATMOSPHERE > CLOUDS > CLOUD MICROPHYSICS > CLOUD DROPLET CONCENTRATION(SIZE), EARTH SCIENCE > ATMOSPHERE > CLOUDS > CLOUD MICROPHYSICS > CLOUD OPTICAL DEPTH(THICKNESS), EARTH SCIENCE > ATMOSPHERE > CLOUDS > CLOUD MICROPHYSICS > CLOUD DROPLET PHASE, EARTH SCIENCE > ATMOSPHERE > CLOUDS > CLOUD PROPERTIES > CLOUD TOP PRESSURE, EARTH SCIENCE > ATMOSPHERE > CLOUDS > CLOUD PROPERTIES > CLOUD TOP TEMPERATURE, EARTH SCIENCE > ATMOSPHERE > CLOUDS > CLOUD PROPERTIES > CLOUD TYPE, EARTH SCIENCE > ATMOSPHERE > CLOUDS > CLOUD PROPERTIES > CLOUD FRACTION, EARTH SCIENCE > ATMOSPHERE > CLOUDS > CLOUD RADIATIVE TRANSFER > CLOUD RADIATIVE FORCING, EARTH SCIENCE > CLIMATE INDICATORS > CRYOSPHERIC INDICATORS > ICE DEPTH(THICKNESS), EARTH SCIENCE > CRYOSPHERE > SEA ICE > ICE DEPTH(THICKNESS),EARTH SCIENCE > CRYOSPHERE > SNOW and ICE > ALBEDO, EARTH SCIENCE > CRYOSPHERE > SNOW and ICE > SNOW and ICE TEMPERATURE, EARTH SCIENCE > LAND SURFACE > LAND TEMPERATURE, EARTH SCIENCE > LAND SURFACE > LAND ALBEDO, EARTH SCIENCE > OCEANS > SEA ICE > ICE DEPTH(THICKNESS),EARTH SCIENCE > TERRESTRIAL HYDROSPHERE > SNOW and ICE > ALBEDO, EARTH SCIENCE > TERRESTRIAL HYDROSPHERE > SNOW and ICE > ICE DEPTH(THICKNESS), ",
    "name": "Extended AVHRR Polar Pathfinder Fundamental Climate Data Record (APPx CDR)",
    "type": "Dataset"
}
```

**Comment by adamml** [comment]
@jyucsiro I think you should (strictly) preface the "type" and "id" keys with @ symbols, such that they become "@type" and "@id". Without, they parse ok but with they are reserved in JSON-LD for the  explicit definitions.

On the identifier and license and keys, the "http://schema.org/" is not required.

As we discussed on the call, it would be nice to split those GCMD keywords into an array...

File: lib/bald/__init__.py [hunk 1]
```diff
@@ -670,8 +670,10 @@ def load(afilepath):
         loader = netCDF4.Dataset
     else:
         raise ValueError('filepath suffix not supported: {}'.format(afilepath))
-    if not os.path.exists(afilepath):
-        raise IOError('{} not found'.format(afilepath))
+    #Disable this check for now to allow URL input
```
Comment (marqh) [inline lib/bald/__init__.py]: do both the hdf and netcdf APIs support loading from URI?

if so, we should just let them do that

is that what you are doing in this case?  does it already work?

Comment (jyucsiro) [reply lib/bald/__init__.py]: I haven't checked with HDF, but it seems to work with the netCDF APIs. 

Yes, that's what I'm doing in this case - both local and from URLs (mainly netCDF files served through thredds).

**Comment by adamml** [comment]
@jyucsiro 

> The issue will be then which namespace the identifier/license properties will come from. This link (https://developers.google.com/search/docs/data-types/dataset) lists license and identifier...

Because you have {"@context": "http://schema.org",...} without the identifier/license properties, you are saying use the schema.org namespace.

**Comment by marqh** [comment]
thank you for all the input, this looks good to go

**Pull request merged**